backend/src/common/model-provider/index.ts

```
@@ -0,0 +1,267 @@
+  private readonly logger = new Logger('ModelProvider');
+  private isDone = false;
+  private responseSubscription: any;
+  private chunkQueue: ChatCompletionChunk[] = [];
+  private resolveNextChunk:
+    | ((value: IteratorResult<ChatCompletionChunk>) => void)
+    | null = null;
+
+  private static instance: ModelProvider | undefined = undefined;
+
+  public static getInstance() {
+    if (this.instance) {
```

Biome

[error] Using this in a static context can be confusing.

this refers to the class.
Unsafe fix: Use the class name instead.


(lint/complexity/noThisInStatic)